utils/tests/key-gen-crypto.bats

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bats
+
+load "std"
+
+# Specialised function for creating BGV data with invalid slot size
+function generate-invalid-bgv-data {
+  genData "${prefix_bgv}_invalid.dat" 4 "BGV" "--columns 3" # 3 in a slot
+  "${encode}" "${prefix_bgv}_invalid.dat" "${prefix_bgv}.info" "BGV" > "${prefix_bgv}_invalid.ptxt"
+}
+
+# Specialised function for creating CKKS data with invalid slot size
+function generate-invalid-ckks-data {
+  genData "${prefix_ckks}_invalid.dat" 4 "CKKS" "--columns 3" # 3 in a slot
+  "${encode}" "${prefix_ckks}_invalid.dat" "${prefix_ckks}.info" "CKKS" > "${prefix_ckks}_invalid.ptxt"
+}
+
+# Only needed for BGV context with bootstrapping
+function generate-bootstrap-data {
+  create-bootstrap-toy-params
+  keyGeneration BGV "${prefix_bootstrap}.params" "${prefix_bootstrap}" "--bootstrap THIN"
+  genData "${prefix_bootstrap}.dat" 12 "BGV" 
+  "${encode}" "${prefix_bootstrap}.dat" "${prefix_bootstrap}.info" "BGV" > "${prefix_bootstrap}.ptxt"
+}
+
+function setup {
+  mkdir -p $tmp_folder
+  cd $tmp_folder
+  print-info-location
+  check_python36
+  check_locations
+  create-bgv-toy-params
+  keyGeneration BGV "${prefix_bgv}.params" "$prefix_bgv"
+  genData "${prefix_bgv}.dat" 12 "BGV" 
+  "${encode}" "${prefix_bgv}.dat" "${prefix_bgv}.info" "BGV" > "${prefix_bgv}.ptxt"
+  create-ckks-toy-params
+  keyGeneration CKKS "${prefix_ckks}.params" "$prefix_ckks"
+  genData "${prefix_ckks}.dat" 12 "CKKS"
+  "${encode}" "${prefix_ckks}.dat" "${prefix_ckks}.info" "CKKS" > "${prefix_ckks}.ptxt"
+}
+
+function teardown {
+  cd -
```

Comment:
do not use this as you cannot guarantee what the previous directory was. santiy check you are in the directory you think you should be in and move to the appropriate one is safer.

Reply:
I can't get an assert on current file location to work -- FYI, this teardown function is used in every existing bats test script so any issue will need to be fixed throughout the lib.